lib/queue.js

```diff
@@ -619,6 +619,8 @@ Queue.prototype.processJob = function(job, renew){
         renew = true;
         lockRenewId = _this.timers.set('lockRenewer', _this.LOCK_RENEW_TIME / 2, lockRenewer);
       }
+      // TODO: if we failed to re-acquire the lock while trying to renew, should we let the job
+      // handler know and cancel the timer?
       return locked;
```

Comment:
When this happens, it must be that the job has been locking the event loop for too long. Could we fail here? but some other worker is already processing the job, should we abort (but we cant do that either)? if we do nothing there is a risk a malfunctioning job keeps being stollen by workers and never complete (or fail)